lib/practice_exercises.rb

```diff
@@ -1,13 +1,35 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n), because we have an array of chars, and we store the char counts in a hash.
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  chars = string.downcase.scan(/\w/)
+  char_counts = Hash.new(0)
+
+  chars.each do |char|
+    char_counts[char] += 1
+  end
+
+  return char_counts.values.all? { |chars| chars.even? } || char_counts.values.one? { |chars| chars.odd? }
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
The time complexity would be O(nm) where n is the number of strings and m is the number of letters in the 1st string.  

Your space complexity is O(m) because you're building `prefix`.